rules/composer/src/ValueObject/AddPackage.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Composer\ValueObject;
+
+use Rector\Composer\Modifier\ComposerModifier;
+use Rector\Composer\Contract\ComposerModifier\ComposerModifierInterface;
+use Webmozart\Assert\Assert;
+
+/**
+ * Only adds package to selected section, if package is already in composer data, nothing happen
+ * @see \Rector\Composer\Tests\ValueObject\AddPackageTest
+ */
+final class AddPackage implements ComposerModifierInterface
+{
+    /** @var string */
+    private $packageName;
+
+    /** @var string */
+    private $version;
+
+    /** @var string */
+    private $section;
+
+    /**
+     * @param string $packageName name of package (vendor/package)
+     * @param string $version target package version (1.2.3, ^1.2, ~1.2.3 etc.)
+     * @param string $section require or require-dev
+     */
+    public function __construct(string $packageName, string $version, string $section = ComposerModifier::SECTION_REQUIRE)
+    {
+        Assert::oneOf($section, [ComposerModifier::SECTION_REQUIRE, ComposerModifier::SECTION_REQUIRE_DEV]);
+
+        $this->packageName = $packageName;
+        $this->version = $version;
+        $this->section = $section;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function modify(array $composerData): array
```

Comment:
Working with array is kind-of unreliable. There is no autocomplete, static analysis nor automated refactoring.

It would be better to work with object `ComposerJson` wich exact API, see
https://tomasvotruba.com/blog/2020/10/31/new-in-symplify-9-composer-json-manipulator-in-object-api/

Reply:
True, didn’t know the package. I can try how it will work here and also in tests

Reply:
I've check this ComposerJson class and I have some reservations:
- I can't add or remove package by calling some method, I have to get packages, then unset key from array and set packages back. Same with replace one package to another. So in my code, I will still work with arrays :(
- it sorts packages always when `setRequire` / `setRequireDev` are called, even if there is no sort-packages in composer.json

Reply:
New methods should be added to Symplify to allow this

Reply:
Ok, so I will prepare PR

Reply:
PR ready for review https://github.com/symplify/composer-json-manipulator/pull/1